osu.Game/Overlays/Toolbar/ToolbarClock.cs

```diff
@@ -85,13 +91,17 @@ protected override void LoadComplete()
 
             clockDisplayMode.BindValueChanged(displayMode =>
             {
-                bool showAnalog = displayMode.NewValue == ToolbarClockDisplayMode.Analog || displayMode.NewValue == ToolbarClockDisplayMode.Full;
-                bool showDigital = displayMode.NewValue != ToolbarClockDisplayMode.Analog;
-                bool showRuntime = displayMode.NewValue == ToolbarClockDisplayMode.DigitalWithRuntime || displayMode.NewValue == ToolbarClockDisplayMode.Full;
+                bool showAnalog = displayMode.NewValue == ToolbarClockDisplayMode.Analog || displayMode.NewValue == ToolbarClockDisplayMode.Full || displayMode.NewValue == ToolbarClockDisplayMode.Full12H;
+                bool showDigital = displayMode.NewValue == ToolbarClockDisplayMode.Digital || displayMode.NewValue == ToolbarClockDisplayMode.DigitalWithRuntime || displayMode.NewValue == ToolbarClockDisplayMode.Full;
+                bool showDigital12H = displayMode.NewValue == ToolbarClockDisplayMode.Digital12H || displayMode.NewValue == ToolbarClockDisplayMode.DigitalWithRuntime12H || displayMode.NewValue == ToolbarClockDisplayMode.Full12H;
+                bool showRuntime = displayMode.NewValue == ToolbarClockDisplayMode.DigitalWithRuntime || displayMode.NewValue == ToolbarClockDisplayMode.DigitalWithRuntime12H || displayMode.NewValue == ToolbarClockDisplayMode.Full || displayMode.NewValue == ToolbarClockDisplayMode.Full12H;
```

Comment:
this is amazing (in a bad way), but not something we can have.

Reply:
Yeah I wasn't really sure the best way to do that, so I just structured it like it was before (which doesn't work great with more options). I think with more than one option (runtime, 12-hour...) it may be better to instead use a single instance of DigitalClockDisplay and have some GUI for the options somewhere. It might be easier to toggle than clicking between 7 (or more) configurations, too.

Reply:
An idea: A dropdown setting. Clicking the clock shows a small dropdown with all the different formats of clock we have. Would be intuitive and faster to config (no spam clicking through formats) and keeps another setting from being added to the long options screen.

Reply:
Good idea, I can work on implementing it this way tomorrow morning

Reply:
This kind of conversation is why we request issues are opened first. Such a design does not exist, and if implemented it is scope creep and also probably will not fit the game's aesthetics and require design work.

The only correct implementation for now that I could see myself merging is automatic detection, as proposed.

https://stackoverflow.com/questions/63261693/is-there-a-datetime-setting-in-net-which-automatically-sets-format-to-12-or-24 may be of help when figuring out how to do this.

Reply:
Also let me add that this is *culture* provided, which isn't 100% accurate.

If this method is used, I'd also proposed a "Prefer 24 hour time" setting alongside the "Prefer original language metadata" setting, for users that prefer 24 hour.

Reply:
Sounds good. If you don't mind my asking though, how should I proceed once I have an implementation? Should I open an issue for discussion first?

Reply:
It's fine to PR as long as it's only adding one setting to the global settings and no new UI.